repos/system_upgrade/el7toel8/actors/migratentp/libraries/ntp2chrony.py

```diff
@@ -0,0 +1,664 @@
+        allowed_networks = filter(lambda n: "ignore" not in self.restrictions[n] and
+                                    "noquery" not in self.restrictions[n] and
+                                    n != ipaddress.ip_network(u"127.0.0.1/32") and
+                                    n != ipaddress.ip_network(u"::1/128"),
+                                  self.restrictions.keys())
+
+        ip_versions = set()
+        conf = ""
+        for network in sorted(allowed_networks, key=lambda n: (n.version, n)):
+            ip_versions.add(network.version)
+            if network.num_addresses > 1:
+                conf += "cmdallow {}\n".format(network)
+            else:
+                conf += "cmdallow {}\n".format(network.network_address)
+
+        if conf:
+            conf = "# Allow remote monitoring.\n" + conf
+            if 4 in ip_versions:
+                conf += "bindcmdaddress 0.0.0.0\n"
+            if 6 in ip_versions:
+                conf += "bindcmdaddress ::\n"
+            conf += "\n"
+
+        return conf
+
+    def get_chrony_conf(self, chrony_keys_path):
+        local_stratum = 0
+        maxdistance = 0.0
+        minsources = 1
+        orphan_stratum = 0
+        logs = []
+
+        for source in self.time_sources:
+            address = source["address"]
+            if address.startswith("127.127.1."):
+                if address in self.fudges and "stratum" in self.fudges[address]:
+                    local_stratum = int(self.fudges[address]["stratum"])
+                else:
+                    local_stratum = 5
+
+        for tos in self.tos_options:
+            if tos[0] == "maxdist":
+                maxdistance = float(tos[1])
+            elif tos[0] == "minsane":
+                minsources = int(tos[1])
+            elif tos[0] == "orphan":
+                orphan_stratum = int(tos[1])
+            else:
+                assert False
+
+        if "clockstats" in self.statistics:
+            logs.append("refclocks");
+        if "loopstats" in self.statistics:
+            logs.append("tracking")
+        if "peerstats" in self.statistics:
+            logs.append("statistics");
+        if "rawstats" in self.statistics:
+            logs.append("measurements")
+
+        conf = "# This file was converted from {}{}.\n".format(
+                    self.ntp_conf_path,
+                    " and " + self.step_tickers_path if self.step_tickers_path else "")
+        conf += "\n"
+
+        if self.ignored_lines:
+            conf += "# The following directives were ignored in the conversion:\n"
+
+            for line in self.ignored_lines:
+                # Remove sensitive information
+                line = re.sub(r"\s+pw\s+\S+", " pw XXX", line.rstrip())
+                conf += "# " + line + "\n"
+            conf += "\n"
+
+        conf += self.get_chrony_conf_sources()
+
+        conf += "# Record the rate at which the system clock gains/losses time.\n"
```

Comment:
Again imho worth considering having the base of config in some variable.

Reply:
There are many strings in the code like this. I don't think it makes sense to put only some in variables.